src/execution_plan/ops/op_merge.c

```diff
@@ -47,14 +47,13 @@ static void _UpdateProperty(Record r, GraphEntity *ge, EntityUpdateEvalCtx *upda
 }
 
 // Perform all ON MATCH updates for all matched records.
-static void _UpdateProperties(OpMerge *op, Record *records) {
+static void _UpdateProperties(OpMerge *op, Record *records, uint record_count) {
 	GraphContext *gc = QueryCtx_GetGraphCtx();
 	uint update_count = array_len(op->on_match);
-	uint record_count = array_len(records);
 
-	// Lock everything.
-	if(!QueryCtx_LockForCommit()) return;
-	{
+	if(op->on_match && update_count && record_count) {
```

Comment:
the check for `op->on_match` seems redundant, as  `uint update_count = array_len(op->on_match);`

Reply:
fixed